rmgpy/solver/base.pyx

```diff
@@ -448,24 +468,30 @@ cdef class ReactionSystem(DASx):
         self.y0 = numpy.zeros(self.neq, numpy.float64)
     
     def set_initial_reaction_thresholds(self):
-        
+
+        # Number of families in RMG
+        len_families = len(getDB('kinetics').families.keys())
+
         # Set unimolecular and bimolecular thresholds as true for any concentrations greater than 0
         numCoreSpecies = len(self.coreSpeciesConcentrations)
         for i in xrange(numCoreSpecies):
-            if self.coreSpeciesConcentrations[i] > 0:
-                self.unimolecularThreshold[i] = True
+            for k in xrange(len_families):
```

Comment:
If we want to vectorize this I think that one of these should work:

```
for i in xrange(numCoreSpecies):
    if self.coreSpeciesConcentrations[i] > 0:
        self.unimolecularThreshold[i,:] = True
```

or

```
self.unimolecularThreshold = np.tile(self.coreSpeciesConcentrations > 0, (len_families, 1)).T
```

If you choose the second one double check that I didn't accidently swap rows and colums--it depends on if self.coreSpeciesConcentrations is a row vector or column vector

If you choose to do this then do the same for bimolecular and trimolecular